src/layers/core/geojson-layer/geojson-layer.js

```diff
@@ -155,64 +120,116 @@ export default class GeoJsonLayer extends Layer {
-        onHover,
-        onClick
+    const {subLayers: {pointFeatures, lineFeatures, polygonFeatures}} = this.state;
+    const {id, getStrokeColor, getStrokeWidth, getFillColor, getHeight} = this.props;
+    const {extruded, wireframe} = this.props;
+
+    let {drawPoints, drawLines, drawPolygons, fillPolygons} = this.props;
+    drawPoints = drawPoints && pointFeatures && pointFeatures.length > 0;
+    drawLines = drawLines && lineFeatures && lineFeatures.length > 0;
+    drawPolygons = drawPolygons && polygonFeatures && polygonFeatures.length > 0;
+    fillPolygons = fillPolygons && polygonFeatures && polygonFeatures.length > 0;
+
+    // Filled Polygon Layer
+    const polygonFillLayer = fillPolygons && new PolygonLayer(Object.assign({}, this.props, {
+      id: `${id}-polygon-fill`,
+      data: polygonFeatures,
+      getPolygon: f => Container.get(f, 'geometry.coordinates'),
+      getHeight,
+      getColor: getFillColor,
+      extruded,
+      wireframe: false,
+      onHover: this._onHoverPolygon,
+      onClick: this._onClickPolygon
+    }));
+
+    // Polygon outline or wireframe
+    let polygonOutlineLayer = null;
+    if (drawPolygons && extruded && wireframe) {
+      polygonOutlineLayer = new PolygonLayer(Object.assign({}, this.props, {
+        id: `${id}-polygon-wireframe`,
+        data: polygonFeatures,
+        getPolygon: f => Container.get(f, 'geometry.coordinates'),
+        getHeight,
+        getColor: getFillColor,
+        extruded: true,
+        wireframe: true,
+        onHover: this._onHoverPolygon,
+        onClick: this._onClickPolygon
       }));
-
-    const pathLayer = pathLayerData &&
-      new PathLayer(Object.assign({}, this.props, {
-        id: `${id}-path-sublayer`,
-        data: pathLayerData,
-        getPaths: this._extractPaths,
+    } else if (drawPolygons) {
+      polygonOutlineLayer = new PathLayer(Object.assign({}, this.props, {
+        id: `${id}-polygon-outline`,
+        data: polygonFeatures,
+        getPaths: f => Container.get(f, 'geometry.coordinates'),
         getColor: getStrokeColor,
         getWidth: getStrokeWidth,
-        getOpacity: getStrokeOpacity,
-        pickable: false,
-        onHover,
-        onClick
-      }));
-
-    const polygonLayer = polygonLayerData &&
-      new PolygonLayer(Object.assign({}, this.props, {
-        id: `${id}-polygon-sublayer`,
-        data: polygonLayerData,
-        getPolygons: this.getFeaturePolygons,
-        getColor: getFillColor,
-        getOpacity: getFillOpacity,
-        pickable: false,
-        onHover,
-        onClick
+        onHover: this._onHoverPolygon,
+        onClick: this._onClickPolygon
       }));
+    }
 
-    return [polygonLayer, pathLayer, pointLayer].filter(Boolean);
+    const lineLayer = drawPolygons && drawLines && new PathLayer(Object.assign({}, this.props, {
+      id: `${id}-line-paths`,
+      data: lineFeatures,
+      getPaths: f => Container.get(f, 'geometry.coordinates'),
```

Comment:
> what was the absolute delta time for the 5x speedup in the bench test?

Creating geometry for all SF zip codes went from 0.1s to 0.02s
- It's a noticeable difference in how quickly the layer pops up when clicking on the layer, and that is not a super large data set. 
- That said, a lot of the speedup (maybe 75%) was in the geometry generation, not in avoiding this particular copy. 

> the general polygon and path layer need to understand geoJson and surface a geoJson feature when hovering

Agreed, those layers shouldn't understand geojson. The way I wanted to make them understand geojson was through having GeoJson layer instantiate them with accessors that understand geojson. I'll review it again to make sure there is no geojson knowledge in `PolygonLayer` and `PathLayer`